internal/commonbuilder/types.go

```diff
@@ -0,0 +1,18 @@
+package commonbuilder
+
+import (
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// CommonBuilderInterface is an interface implemented by all builders.
+type CommonBuilderInterface interface {
+	GetClient() goclient.Client
+	GetDefinition() goclient.Object
+	GetErrorMsg() string
+	GetKind() string
```

Comment:
IMO, this one is a nice to have but not a deal breaker. It really only adds a little bit of context to the error messages but we could really let the user handle that in their own logging/error checking